src/frontends/tensorflow_common/src/op/segment_sum.cpp

```diff
@@ -45,6 +46,14 @@ OutputVector translate_segment_sum_op(const NodeContext& node) {
     auto num_indices = make_shared<v0::Squeeze>(segment_ids_shape, squeeze_axis);
     auto indices = make_shared<v4::Range>(const_zero, num_indices, const_one, indices_type);
 
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(data.get_node_shared_ptr());
+    if (complex_type_mark) {
+        data = complex_type_mark->input_value(0);
+        auto emb_segment_sum_complex = make_shared<v3::EmbeddingSegmentsSum>(data, indices, segment_ids, num_segments);
```

Comment:
@hegdeadithyak, add `ComplexTypeMark` to output of `emb_segment_sum_complex` because output will have complex type. See other operations with complex type support